### PR TITLE
Add users

### PR DESCRIPTION
**Pull request by caward12**
- create login page
- create create account page
- added sessions controller 
- added login/logout links

### CODE REVIEW

File: .rspec [hunk 1]
```diff
@@ -1,3 +1,3 @@
 --color
 --require spec_helper
---format NyanCatFormatter
+--format NyanCatMusicFormatter
```
Comment (NessEFC) [inline .rspec]: This is an important option to add. Nice work!

Comment (lukyans) [inline .rspec]: YES

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      session[:user_id] = @user.id
+      flash[:success] = "Logged in as #{@user.username}"
+      redirect_to dashboard_path
+    else
+      flash[:message] = "Try again"
+      render :new
+    end
+  end
+
+    def dashboard
```
Comment (BethKnight1234) [inline app/controllers/users_controller.rb]: Dashboard = show

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      session[:user_id] = @user.id
+      redirect_to dashboard_path
+    else
+      render :new
+    end
+  end
+
+  def destroy
+    session.clear
+    flash[:message] = "Successfully logged out. Bye Felicia."
```
Comment (NessEFC) [inline app/controllers/sessions_controller.rb]: Can we change "Felicia" to the user's username?

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+class User < ApplicationRecord
+  has_secure_password
+
```
Comment (NessEFC) [inline app/models/user.rb]: Validation tests added in Sergey's next PR.

**Pull request merged**